pandas/tests/series/test_operators.py

```diff
@@ -993,15 +991,44 @@ def test_operators_timedelta64_with_timedelta_invalid(self, scalar_td):
             td1 * scalar_td
         with tm.assert_raises_regex(TypeError, pattern):
             scalar_td * td1
-        with tm.assert_raises_regex(TypeError, pattern):
-            td1 // scalar_td
-        with tm.assert_raises_regex(TypeError, pattern):
-            scalar_td // td1
         with tm.assert_raises_regex(TypeError, pattern):
             scalar_td ** td1
         with tm.assert_raises_regex(TypeError, pattern):
             td1 ** scalar_td
 
+    @pytest.mark.parametrize('scalar_td', [
+        timedelta(minutes=5, seconds=4),
+        pytest.param(Timedelta('5m4s'),
+                     marks=pytest.mark.xfail(reason="Timedelta.__floordiv__ "
+                                                    "bug GH#18846")),
+        Timedelta('5m4s').to_timedelta64()])
+    def test_timedelta_rfloordiv(self, scalar_td):
+        # GH#18831
+        td1 = Series([timedelta(minutes=5, seconds=3)] * 3)
+        td1.iloc[2] = np.nan
+        result = scalar_td // td1
+        expected = Series([1, 1, np.nan])
+        tm.assert_series_equal(result, expected)
+
+    @pytest.mark.parametrize('scalar_td', [
+        timedelta(minutes=5, seconds=4),
+        Timedelta('5m4s'),
+        Timedelta('5m4s').to_timedelta64()])
+    def test_timedelta_floordiv(self, scalar_td):
+        # GH#18831
+        td1 = Series([timedelta(minutes=5, seconds=3)] * 3)
+        td1.iloc[2] = np.nan
+
+        result = td1 // scalar_td
+        expected = Series([0, 0, np.nan])
+        tm.assert_series_equal(result, expected)
+
+        # We can test __rfloordiv__ using this syntax,
```

Comment:
shouldn't this be in the other test? (as the other one is rfloordiv), but this seems to work?

Reply:
It's kind of a man without a country regardless.  I didn't want to put it in the rfloordiv test because that was xfailed so putting it there would be akin to not running it at all.  But you're right it doesn't _really_ belong in the floordiv test either.  I'm open to suggestions.

Reply:
then make another test for it